aten/src/ATen/native/cpu/BlasKernel.cpp

```diff
@@ -83,6 +83,36 @@ void gemm_notrans_(
   }
 }
 
+template <typename scalar_t, typename opmath_t>
+typename std::enable_if<std::is_same<scalar_t, at::BFloat16>::value, void>::type
+gemm_notrans_(
+    int64_t m,
+    int64_t n,
+    int64_t k,
+    opmath_t alpha,
+    const scalar_t* a,
```

Comment:
This doesn't need to be a template, and there's no need for `enable_if` anywhere in this file
```suggestion
    float alpha,
    const c10::BFloat16* a,
```

Reply:
I used template with `enable_if` so the code changes will be smaller while supporting `Half`.

Reply:
Okay, then why not do `is_same<scalar_t, opmath_t>`?

Reply:
Good idea.